compiler/rustc_parse_format/src/lib.rs

```diff
@@ -856,6 +858,19 @@ fn find_width_map_from_snippet(
         return (vec![], true);
     }
 
+    // Macros like `println` add a newline at the end. That technically doens't make them "literals" anymore, but it's fine
+    // since we will never need to point our spans there, so we lie about it here by ignoring it.
+    // Since there might actually be newlines in the source code, we need to normalize away all trailing newlines.
+    let input_no_nl = input.trim_end_matches('\n');
```

Comment:
Do we only want to trim only one `\n` off the literal to deal with the `println!` case, then check strict literal equality? Or am I missing something more subtle here?

Reply:
You are just like I was when I tried this at first.
If we have `format!("uwu\n")`, then trimming it off the input will cause a mismatch, as it's still present in the snippet.
Doing "only trim it when the snippet doesn't end with a newline" also doesn't work because it could be `println!("uwu\n")`.
We would need to actually count the trailing newlines and match and only trim it off when the input has one more, but doing it like I did is a little simpler.

I will explain this better in the comment.

Reply:
Ah, yeah, I guess the important part is that we can't distinguish when the newline is added by the macro or already present in the escaped literal. Oh well.